### DISCUSSION TITLE
Best Persistence Practice

### DISCUSSION THREAD

**Discussion by Enxyphered** [angularsen/UnitsNet#1197] [Q&A]
I can imagine a few ways of using this to persist data. Specifically Entities using EntityFramework. I was wondering what the best and most recommended practice for creating Entities using this library. For example lets say I have a "InventoryItem" entity with a BaseUnitInfo property and let's say I have RecievingItem Entity with an IQuantity that needs to be saved to the database.  I was hoping someone could give me any tips or anything that a new user of this library would miss when starting to tackle such problems.

Thanks in advance,
Shaamil.

**Comment by Enxyphered** [comment]
My initial guess would be that we have our own UnitType enum that lists all the types of units relevant to our program. For each UnitType we also have Enum for the Units under that UnitType. Now we can store the UnitType and Unit as integers in our Entities. Then we can use UnitType and Unit to convert to UnitInfo in this Library and subsequently have a function that takes UnitType, Unit and Magnitude as parameters to return IQuantity. I feel like that's somehow too convoluted and a lot of code.

**Reply by angularsen** [reply]
Hi, the only supported persistence is JSON and XML. You could take a look at it for inspiration. 
https://github.com/angularsen/UnitsNet/wiki/Serializing-to-JSON,-XML-and-more

Persistence is not trivial, in particular handling breaking changes such as renamed/removed units and unit abbreviations. If you have a finite set of known units you need to persist, you are probably best off using your own enums or strings and mapping from/to UnitsNet types as you touched on already.

Hope this helps.
